extension/src/execute.rs

```diff
@@ -30,16 +30,16 @@ unsafe extern "C" fn execute_ex(execute_data: *mut sys::zend_execute_data) {
     };
 
     // Run the upstream function and record the duration.
-    let start = get_unix_timestamp_micros();
+    let start = Instant::recent();
```

Comment:
One problem is that you only update on function end. I don’t know how realistic it is to never call a small function that returns, but maybe a good way to solve this is to have a call stack depth % 20 or so and call update() every once in a while.

That can be a stack local variable - right now.

Reply:
I'll take a look at options. Thanks for the review @LionsAd !